src/diagnoses/diagnoses.component.tsx

```diff
@@ -0,0 +1,72 @@
+import { InlineLoading, InlineNotification } from '@carbon/react';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { usePatientDiagnosis } from './diagnoses.resource';
+import {
+  StructuredListWrapper,
+  StructuredListHead,
+  StructuredListRow,
+  StructuredListCell,
+  StructuredListBody,
+  Layer,
+} from '@carbon/react';
+import styles from './diagnoses.scss';
+import EmptyState from '../components/empty-state.component';
+
+type PatientDiagnosesProps = {
+  patientUuid: string;
+  encounterUuid: string;
+};
+
+const PatientDiagnoses: React.FC<PatientDiagnosesProps> = ({ encounterUuid, patientUuid }) => {
+  const { diagnoses, isLoading, error, showDiagnosesFromVisit } = usePatientDiagnosis(encounterUuid);
+
+  const { t } = useTranslation();
+
+  if (!showDiagnosesFromVisit) return null;
+
+  if (isLoading)
+    return (
+      <InlineLoading
+        iconDescription="Loading"
+        description={t('loadingDiagnoses', 'Loading Diagnoses ...')}
+        status="active"
+      />
+    );
+
+  if (error)
+    return <InlineNotification kind="error" subtitle={t('diagnosesError', 'Error loading diagnoses')} lowContrast />;
+
+  if (!diagnoses.length)
+    return (
+      <EmptyState
+        title={t('diagnoses', 'Diagnoses')}
+        message={t('noDiagnoses', "No diagnoses for this patient's visit")}
+      />
+    );
+
+  return (
+    <Layer className={styles.diagnosesContainer}>
+      <StructuredListWrapper>
+        <StructuredListHead>
+          <StructuredListRow head>
+            <StructuredListCell head>
+              {t('diagnoses', 'Diagnoses')} {`(${diagnoses.length})`}
+            </StructuredListCell>
+            <StructuredListCell head>{t('status', 'Status')}</StructuredListCell>
+          </StructuredListRow>
+        </StructuredListHead>
+        <StructuredListBody>
+          {diagnoses.map(({ certainty, text }) => (
+            <StructuredListRow>
```

Comment:
Any time you call `map()` in a React component, the component _must_ have a key.